gax-java/gax/src/test/java/com/google/api/gax/core/GaxPropertiesTest.java

```diff
@@ -57,10 +58,94 @@ public void testGaxVersion() {
+    }
+
+    if (Strings.isNullOrEmpty(originalJavaVendorVersion)) {
+      System.clearProperty("java.vendor.version");
+    } else {
+      System.setProperty("java.vendor.version", originalJavaVendorVersion);
+    }
+  }
+
+  @Test
+  public void testGetJavaRuntimeInfo_graalVM() {
+
+    System.setProperty("java.version", "17.0.3");
+    System.setProperty("java.vendor", "GraalVM Community");
+    System.setProperty("java.vendor.version", "GraalVM CE 22.1.0");
+
+    String runtimeInfo = GaxProperties.getRuntimeVersion();
+    assertEquals("17.0.3__GraalVM-Community__GraalVM-CE-22.1.0", runtimeInfo);
+  }
+
+  @Test
+  public void testGetJavaRuntimeInfo_temurin() {
+    System.setProperty("java.version", "11.0.19");
+    System.setProperty("java.vendor", "Eclipse Adoptium");
+    System.setProperty("java.vendor.version", "Temurin-11.0.19+7");
+
+    String runtimeInfo = GaxProperties.getRuntimeVersion();
+    assertEquals("11.0.19__Eclipse-Adoptium__Temurin-11.0.19-7", runtimeInfo);
+  }
+
+  @Test
+  public void testGetJavaRuntimeInfo_coretto() {
+    System.setProperty("java.version", "11.0.19");
+    System.setProperty("java.vendor", "Amazon.com Inc.");
+    System.setProperty("java.vendor.version", "Corretto-11.0.19.7.1");
+
+    String runtimeInfo = GaxProperties.getRuntimeVersion();
+    assertEquals("11.0.19__Amazon.com-Inc.__Corretto-11.0.19.7.1", runtimeInfo);
+  }
+
   @Test
-  public void testGetVersion_nativeImage() {
-    System.setProperty(PROPERTY_IMAGE_CODE_KEY, PROPERTY_IMAGE_CODE_VALUE_RUNTIME);
-    String javaVersion = GaxProperties.getJavaVersion();
-    assertTrue(javaVersion.endsWith("-graalvm"));
+  public void testGetJavaRuntimeInfo_oracle() {
+    System.setProperty("java.version", "20.0.1");
+    System.setProperty("java.vendor", "Oracle Corporation");
+    // case where java.vendor.version is null
+    System.clearProperty("java.vendor.version");
+
+    String runtimeInfo = GaxProperties.getRuntimeVersion();
+    assertEquals("20.0.1__Oracle-Corporation", runtimeInfo);
+  }
+
+  @Test
+  public void testGetJavaRuntimeInfo_nullVendorVersion() {
+    System.setProperty("java.version", "20.0.1");
+    // case where java.vendor and java.vendor.version is null
+    System.clearProperty("java.vendor");
+    System.clearProperty("java.vendor.version");
+
+    String runtimeInfo = GaxProperties.getRuntimeVersion();
+    assertEquals("20.0.1", runtimeInfo);
+  }
+
+  @Test
+  public void testGetJavaRuntimeInfo_nullJavaVersion() {
+    // We don't expect this case to happen, however we don't want the method to fail when it really
+    // happens.
+    System.clearProperty("java.version");
+    // case where java.vendor and java.vendor.version is null
+    System.setProperty("java.vendor", "oracle");
+    System.setProperty("java.vendor.version", "20.0.1");
+
+    String runtimeInfo = GaxProperties.getRuntimeVersion();
+    assertEquals("null__oracle__20.0.1", runtimeInfo);
```

Comment:
Is this expected to have a String `null` in the version info?

Reply:
yes. If the system property does not exist, we are assigning it a null value.

Reply:
We are assigning it a "null" as String.